### PR TITLE
N7s [ETD-80] extract ccsds mode into discrete packetizer

### PR DESCRIPTION
**Pull request by DamianGzik**

### CODE REVIEW

File: src/Packetizer/CCSDSPacketizer.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                           const Packetizer_PacketType packetType,
+                           const uint16_t source,
+                           const uint16_t destination,
+                           uint8_t* const packetPointer,
+                           const size_t dataOffset,
+                           const size_t dataSize)
+{
+    // Unused in this implementation
+    (void)source;
+    (void)dataOffset;
```
Comment (Lurkerpas) [inline src/Packetizer/CCSDSPacketizer.c:50]: dataOffset is used in assert, right? Or does the compiler return a warning in release build?

File: tests/Packetizer/CCSDSPacketizerTests.cc [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    auto resultSize = CCSDS_Packetizer_packetize(&packetizer,
+                                                 Packetizer_PacketType_Telemetry,
+                                                 0,
+                                                 destination,
+                                                 packetData,
+                                                 SPACE_PACKET_PRIMARY_HEADER_SIZE,
+                                                 dataSize);
+
+    CHECK_EQUAL(packetSize, resultSize);
+    CHECK_EQUAL(0b00001010, packetData[0]);
```
Comment (Lurkerpas) [inline tests/Packetizer/CCSDSPacketizerTests.cc:38]: These tests are nice, because they have a "fixed" reference. However, they are hard to understand by a human. Would it be possible to add at least one test, which first packetizes and then depacketizes a packet, and checks whether the output data and meta is equal to the one on the input?

File: src/Packetizer/CCSDSPacketizer.c (hunk 1 above)
Comment (DamianGzik) [reply src/Packetizer/CCSDSPacketizer.c:50]: dataOffset is used in assert later on

File: src/Broker/Broker.c [hunk 3]
```diff
@@ -49,9 +50,9 @@ Broker_initialize_packetizers_functions()
     packetizers_functions[PACKETIZER_DEFAULT].packetize = &Packetizer_packetize;
     packetizers_functions[PACKETIZER_DEFAULT].depacketize = &Packetizer_depacketize;
     packetizers_functions[PACKETIZER_CCSDS].headerSize = SPACE_PACKET_PRIMARY_HEADER_SIZE;
```
[the remark this answers is not in the document]
Comment (DamianGzik) [reply src/Broker/Broker.c]: Done. PACKETIZER_CCSDS reverted as it was. Packetizers PACKETIZER_STRICT_CCSDS function added with new CCSDS_Packetizer.

File: src/Packetizer/CCSDSPacketizer.c (hunk 1 above)
Comment (Lurkerpas) [reply src/Packetizer/CCSDSPacketizer.c:50]: OK, do we need then the (void)dataOffset; ?

Comment (DamianGzik) [reply src/Packetizer/CCSDSPacketizer.c:50]: Yes we do. If it is only in assert and not used later in code (referenced) then there is build error
image

File: src/Broker/Broker.c [hunk 4]
```diff
@@ -52,6 +53,10 @@ Broker_initialize_packetizers_functions()
     packetizers_functions[PACKETIZER_CCSDS].init = &Packetizer_init;
     packetizers_functions[PACKETIZER_CCSDS].packetize = &Packetizer_packetize;
     packetizers_functions[PACKETIZER_CCSDS].depacketize = &Packetizer_depacketize;
+    packetizers_functions[PACKETIZER_STRICT_CCSDS].headerSize = SPACE_PACKET_PRIMARY_HEADER_SIZE;
```
[the remark this answers is not in the document]
Comment (Lurkerpas) [reply src/Broker/Broker.c]: Code changed.

**Review by Lurkerpas — DISMISSED** [review]

File: tests/Broker/BrokerTests.cc [hunk 5]
```diff
@@ -66,14 +66,14 @@ TEST(Broker, InvalidCrc)
 
     Broker_register_error_callback(&testErrorDetectedCallback);
 
-    uint8_t data[] = { 0, 0, 0, 0, 0, 1, 0, 0, 0, 0 };
+    uint8_t data[] = { 0, 0, 0, 0, 255, 255, 255, 255, 0, 0, 0, 0 };
```
Comment (Lurkerpas) [inline tests/Broker/BrokerTests.cc:69]: Why this change?

Comment (psurm) [reply tests/Broker/BrokerTests.cc:69]: I guess the test previously expected 'strict CCSDS' behaviour in some places.  The length field for non strict CCSDS takes 4 bytes and 'zero' value is represented by 0xFFFFFFFF because it is always decremented by 1.

`size_t
readPacketDataLength(const uint8_t* const packetPointer)
{
    const uint32_t packetSize =
            ((uint32_t)(packetPointer[4] << 24u) | packetPointer[5] << 16u | packetPointer[6] << 8u | packetPointer[7]);
    const uint32_t zeroPacketSize = 0xffffffff;
    if(packetSize == zeroPacketSize) {
        return 0;
    }
    return (size_t)packetSize + 1;
}
`

**Review by Lurkerpas — APPROVED** [review]
